test/integration/testExecute.js

```diff
@@ -137,6 +137,25 @@ describe('Execute test', function () {
       done
     );
   });
+
+  it('testDescribeOnly', function (done) {
+    const selectWithDescribeOnly = 'SELECT 1.0::NUMBER(30,2) as C1, 2::NUMBER(38,0) AS C2, \'t3\' AS C3, 4.2::DOUBLE AS C4, \'abcd\'::BINARY(8388608) AS C5, true AS C6';
+
+    function executeQueryAndVerifyResultDependOnDescribeOnly(describeOnly) {
+      connection.execute({
+        sqlText: selectWithDescribeOnly,
+        describeOnly: describeOnly,
+        complete: (err, stmt, rows) => {
+          assert.strictEqual(stmt.getColumns().length, 6);
+          //Empty rowset in response
+          assert.strictEqual(rows.length, describeOnly ? 0 : 1);
```

Comment:
How can the user access data returned by GS for DESCRIBE, since there are no rows returned? Should we have test for that as well?

Reply:
DESCRIBE also returns only metadata, rows are empty.

Reply:
As we found out on site - the DESCRIBE statement returns rows, regardless of a value set to the describeOnly parameter (e.g. when sql txt of the above test is changed to: "DESCRIBE DATABASE TESTDB").
This happens when selecting data from a real table as well (e.g. "SELECT ... FROM REAL_TABLE_NAME").

Knowing that, we should probably investigate further what is the expected behaviour of this flag, to be able to determine wether our tests are covering such cases.